src/test/java/codeu/controller/ProfileServletTest.java

```diff
@@ -0,0 +1,131 @@
+  private HttpServletResponse mockResponse;
+  private RequestDispatcher mockRequestDispatcher;
+  private MessageStore mockMessageStore;
+  private UserStore mockUserStore;
+
+  @Before
+  public void setup() {
+    profileServlet = new ProfileServlet();
+
+    mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockSession = Mockito.mock(HttpSession.class);
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+
+    mockResponse = Mockito.mock(HttpServletResponse.class);
+    mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/profile.jsp"))
+        .thenReturn(mockRequestDispatcher);
+
+    mockMessageStore = Mockito.mock(MessageStore.class);
+    profileServlet.setMessageStore(mockMessageStore);
+
+    mockUserStore = Mockito.mock(UserStore.class);
+    profileServlet.setUserStore(mockUserStore);
+  }
+
+  @Test
+  public void testDoGet() throws IOException, ServletException {
+    User fakeUser = new TestUserBuilder().withName("test_user").build();
+    Mockito.when(mockRequest.getRequestURI()).thenReturn("/users/test_user");
+    Mockito.when(mockUserStore.getUser("test_user"))
+        .thenReturn(fakeUser);
+
+    String fakeUsername = "test_user";
+    Mockito.when(mockRequest.getAttribute("username")).thenReturn("test_user");
+
+    List<Message> fakeMessagesByUser = new ArrayList<>();
+    fakeMessagesByUser.add(new TestMessageBuilder().withAuthorId(fakeUser.getId()).build());
+    Mockito.when(mockMessageStore.getMessagesByUser(fakeUser.getId()))
+        .thenReturn(fakeMessagesByUser);
+
+    profileServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest).setAttribute("messagesByUser", fakeMessagesByUser);
+    Mockito.verify(mockRequest).setAttribute("username", fakeUsername);
+    Mockito.verify(mockRequest).setAttribute("user", fakeUser);
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void testDoPost_UserNotLoggedIn() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn(null);
+
+    profileServlet.doPost(mockRequest, mockResponse);
+
+    Mockito.verify(mockMessageStore, Mockito.never()).addMessage(Mockito.any(Message.class));
+    Mockito.verify(mockResponse).sendRedirect("/login");
+  }
+
+  @Test
+  public void testDoPost_InvalidUser() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("test_user");
+    Mockito.when(mockUserStore.getUser("test_user")).thenReturn(null);
+
+    profileServlet.doPost(mockRequest, mockResponse);
+
+    Mockito.verify(mockMessageStore, Mockito.never()).addMessage(Mockito.any(Message.class));
+    Mockito.verify(mockResponse).sendRedirect("/login");
+  }
+
+  @Test
+  public void testDoPost_CleansHtmlContent() throws IOException, ServletException {
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
```

Comment:
This is already in `setup` so you don't also need to have it here.  The `setup` method will always run before each test (because it's annotated with `@Before`).